### PR TITLE
Feature/neigbor list getitem

### LINKED ISSUE #225: Support __getitem__ on NeighborLists
**Is your feature request related to a problem? Please describe.**
It takes a lot of code just to extract `(i, j)` pairs from a `NeighborList` object.

```python
linkcell = freud.locality.LinkCell(box, r_max)
linkcell.compute(box, particles)
for i, j in zip(linkcell.nlist.index_i, linkcell.nlist.index_j):
    print(i, j)
```

**Describe the solution you'd like**
If the `__getitem__` function were implemented for `NeighborList`, that code would just become:
```python
for i, j in linkcell.nlist:  # Much simpler
    print(i, j)
```

The array `nlist[:]` would be of shape `(Nbonds, 2)` with `dtype=np.uint64`.

The functions for `index_i` and `index_j` could be greatly simplified by returning `self[:, 0]` and `self[:, 1]` for `i` and `j`, respectively.

**Additional context**
Ran into this issue again today with @tcmoore3.

### PR DESCRIPTION
**Pull request by klywang**
## Description

Resolves #225 
Created a setup funciton for Neightborlist tests

## Motivation and Context
Reduces amount of code required to extract (i, j) from NeighborList

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: doc/source/credits.rst [hunk 1]
```diff
@@ -212,7 +217,7 @@ various order parameters, under the following license::
     Copyright (c) 2016 The Regents of the University of Michigan
 
     Permission is hereby granted, free of charge, to any person obtaining a copy
-    of this software and associated documentation files (the "Software"), to deal
+    of this software and associated documentation files (the "Software"), to dealhbor
```
Comment (bdice) [inline doc/source/credits.rst]: @k-l-wang Can you fix this? 😆

File: tests/test_locality_NeighborList.py [hunk 2]
```diff
@@ -162,5 +144,57 @@ def test_from_arrays(self):
...
+
+    def test_nl_size(self):
+
+        self.setup_nl()
+
+        np.random.seed(0)
+
+        self.cl.compute(self.fbox, self.points, self.points)
+        assert len([x for x in self.cl.nlist]) == len(self.cl.nlist.index_i)
+        assert len([x for x in self.cl.nlist]) == len(self.cl.nlist.index_j)
```
Comment (bdice) [inline tests/test_locality_NeighborList.py]: I think this is equivalent to saying `assert len(self.cl.nlist) == len(self.cl.nlist.index_j)`. Can you make that change to simplify the test?

**Review by bdice — CHANGES_REQUESTED** [review]
Thanks for the pull request! 😄 This is a nice feature and I am excited to use this in my code. Please see the change requests that I have written and let me know if you have any questions!

File: doc/source/credits.rst (hunk 1 above)
Comment (vyasr) [reply doc/source/credits.rst]: I fixed this typo!

**Comment by vyasr** [comment]
@bdice I fixed the minor typos. Feel free to merge once you go through another round of review (I'll leave it to you since you requested changes).

**Review by bdice — APPROVED** [review]

**Pull request merged**